commands/rank.js

```diff
@@ -0,0 +1,141 @@
+const Discord = require('discord.js');
+const Enmap = require("enmap");
+
+const ALLOWED_RANKS = [
+  "Iron1",
+  "Iron2",
+  "Iron3",
+  "Bronze1",
+  "Bronze2",
+  "Bronze3",
+  "Silver1",
+  "Silver2",
+  "Silver3",
+  "Gold1",
+  "Gold2",
+  "Gold3",
+  "Platinum1",
+  "Platinum2",
+  "Platinum3",
+  "Diamond1",
+  "Diamond2",
+  "Diamond3",
+  "Immortal1",
+  "Immortal2",
+  "Immortal3",
+  "Radiant"
+];
+
+exports.run = async (client, message, args, level) => {
+  client.ranks = new Enmap({name: "ranks"});
+  const channel = message.channel;
+
+  const subcommand = args[0];
+
+  if(!subcommand) {
+    return sendHelp(channel);
+  }
+
+  switch(subcommand) {
+    case "set":
+      const rank = args[1];
+
+      if(!rank || !ALLOWED_RANKS.includes(rank)) {
+        // invalid rank, send allowed ranks
+        return sendRanks(channel);
+      } else {
+        const key = message.author.id;
+
+        // save rank
+        client.ranks.set(key, {
```

Comment:
do we save the key (id) in the obj so its easier to readout for leaderboard?

Reply:
ya i only did it cuz `client.ranks.array()` only returns values, so it doesn't have the keys.  I couldn't find a "getEntries" function but I didn't look very hard